routers/web/repo/repo.go

```diff
@@ -622,26 +622,47 @@ func SearchRepo(ctx *context.Context) {
 		}
 	}
 
-	var err error
+	getErrorStr := func(ctx *context.Context, err error, fallbackStr string) string {
+		if ctx.Doer == nil || !ctx.Doer.IsAdmin {
+			return fallbackStr
+		}
+		return fmt.Sprintf("%s: %v", fallbackStr, err)
+	}
+
+	// To improve performance when only the count is requested
+	if ctx.FormBool("count_only") {
+		count, err := repo_model.CountRepository(ctx, opts)
+		if err != nil {
+			log.Error("CountRepository: %v", err)
+			ctx.JSON(http.StatusInternalServerError, api.SearchError{
+				OK:    false,
+				Error: getErrorStr(ctx, err, "CountRepository internal error"),
+			})
+			return
+		}
+		ctx.SetTotalCountHeader(count)
+		return
+	}
+
 	repos, count, err := repo_model.SearchRepository(ctx, opts)
 	if err != nil {
+		log.Error("SearchRepository: %v", err)
 		ctx.JSON(http.StatusInternalServerError, api.SearchError{
 			OK:    false,
-			Error: err.Error(),
+			Error: getErrorStr(ctx, err, "SearchRepository internal error"),
```

Comment:
`getErrorStr` makes the code more hacky. Would the code be filled with a lot of `getErrorStr` in every handler?

And I would ask the question a third time:

> And, I don't see why it should use api.SearchError here, is the response really handled by frontend JS?

If I didn't misread the code, the frontend code NEVER handles the response error. It doens't need to respond anything here.

Reply:
->  Only do counting when count_only=true for repo dashboard #29884 

What do you think about this? It is simple enough and could be backported.